internal/aws/s3/s3.go

```diff
@@ -170,16 +186,32 @@ func Upload(bucketName string, content []byte) (string, error) {
 
 // RainBucket returns the name of the rain deployment bucket in the current region
 // and asks the user if they wish it to be created if it does not exist
-// unless forceCreation is true, then it will not ask
+// unless forceCreation is true, then it will not ask.
+// If a blank string is passed in, we look for a parameter store key named "rain-bucket".
+// If that doesn't exist, we use "rain-artifacts-accountid-region".
+// If a non-blank string is passed in, we create that bucket if it doesn't exist.
 func RainBucket(forceCreation bool) string {
 	accountID, err := sts.GetAccountID()
 	if err != nil {
 		panic(fmt.Errorf("unable to get account ID: %w", err))
 	}
 
+	// --bucket-name is passed in as an arg to various commands
 	bucketName := BucketName
+
 	if bucketName == "" {
-		bucketName = fmt.Sprintf("rain-artifacts-%s-%s", accountID, aws.Config().Region)
+		storedName, err := ssm.GetParameter(RAIN_BUCKET_SSM_KEY)
+		if err != nil {
+			// This is expected if the key is not found
+			config.Debugf("Could not get %s: %v", RAIN_BUCKET_SSM_KEY, err)
+		}
+		if storedName != "" {
+			config.Debugf("Found bucket name in parameter store: %s", storedName)
+			bucketName = storedName
+		} else {
+			config.Debugf("Bucket name not found in parameter store")
+			bucketName = fmt.Sprintf("rain-artifacts-%s-%s", accountID, aws.Config().Region)
```

Comment:
In light of the issue being addressed, perhaps this could include a UUID?

Reply:
Could that cause backwards compatibility issues for customers who already bootstrapped with the original name?

Reply:
I tried a few ideas on a branch and I found that it introduced a bit too much complexity. Since we fail if the bucket is not in the same account, I think it's Ok to leave it as a predictable name.